app/src/main/kotlin/com/simplemobiletools/smsmessenger/activities/MainActivity.kt

```diff
@@ -318,6 +337,99 @@ class MainActivity : SimpleActivity() {
         startAboutActivity(R.string.app_name, licenses, BuildConfig.VERSION_NAME, faqItems, true)
     }
 
+    private fun tryToExportMessages() {
+        if (isQPlus()) {
+            ExportMessagesDialog(this, config.lastExportPath, true) { file ->
+
+                Intent(Intent.ACTION_CREATE_DOCUMENT).apply {
+                    type = EXPORT_MIME_TYPE
+                    putExtra(Intent.EXTRA_TITLE, file.name)
+                    addCategory(Intent.CATEGORY_OPENABLE)
+                    startActivityForResult(this, PICK_EXPORT_FILE_INTENT)
+                }
+            }
+        } else {
+            handlePermission(PERMISSION_WRITE_STORAGE) {
+                if (it) {
+                    ExportMessagesDialog(this, config.lastExportPath, false) { file ->
+                        getFileOutputStream(file.toFileDirItem(this), true) { outStream ->
+                            exportMessagesTo(outStream)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun exportMessagesTo(outputStream: OutputStream?) {
+        ensureBackgroundThread {
+            toast(R.string.exporting)
+            smsExporter.exportMessages(outputStream) {
+                toast(
+                    when (it) {
+                        MessagesExporter.ExportResult.EXPORT_OK -> R.string.exporting_successful
+                        else -> R.string.exporting_failed
+                    }
+                )
+            }
+        }
+    }
+
+    private fun tryImportMessages() {
+        if (isQPlus()) {
+            Intent(Intent.ACTION_GET_CONTENT).apply {
+                addCategory(Intent.CATEGORY_OPENABLE)
+                type = EXPORT_MIME_TYPE
+                startActivityForResult(this, PICK_IMPORT_SOURCE_INTENT)
+            }
+        } else {
+            handlePermission(PERMISSION_READ_STORAGE) {
+                if (it) {
+                    importEvents()
+                }
+            }
+        }
+    }
+
+    private fun importEvents() {
+        FilePickerDialog(this) {
+            showImportEventsDialog(it)
+        }
+    }
+
+    private fun showImportEventsDialog(path: String) {
+        ImportMessagesDialog(this, path) { refresh ->
+            if (refresh) {
+                runOnUiThread {
+
```

Comment:
should something happen here?

Reply:
@KryptKode check if this callback is needed at all please, if not, just remove it altogether

Reply:
Okay. Will do